app/Jobs/TranscodeVideo.php

```diff
@@ -123,7 +123,6 @@ public function failed(?Throwable $exception): void
         $localDisk->deleteDirectory($this->getFfmpegTempDirectory());
 
         if (!$this->oldVersionNumber) {
-            MediaStorage::ORIGINALS->getDisk()->delete($this->originalFilePath);
             $this->version->delete();
             $versionNumber = $this->version->number - 1;
         } else {
```

Comment:
```suggestion
        } else {
            // Restoring an old version has failed. It will not get a higher version number, but keep its old one.
```